samples/doc-snippets/src/main/java/com/example/application/ActivityAgentMore.java

```diff
@@ -195,6 +198,37 @@ public HttpResponse askGrouped(Request request) {
  }
  // end::stream-endpoint[]
 
+ @ComponentId("activity-agent")
+ public class ActivityAgentWithErrorHandling extends Agent {
+     
+     private static final String SYSTEM_MESSAGE =
+     """
+        You are an activity agent. Your job is to suggest activities in the
+        real world. Like for example, a team building activity, sports, an
+        indoor or outdoor game, board games, a city trip, etc.
+        """.stripIndent();
+        
+        // tag::error-handling[]
+        public Effect<String> query(String message) {
+            return effects()
+            .systemMessage(SYSTEM_MESSAGE)
+            .userMessage(message)
+            .onFailure(exception -> {
+                // Handle different types of exceptions with appropriate fallback responses
+                return switch (exception) {
+                  case RateLimitException exc ->
+                    "Rate limit exceeded: please try again later";
+                  case ModelTimeoutException exc ->
+                    "Request timeout: the service is experiencing delays";
+                  case ToolCallExecutionException exc ->
+                    "Tool error: unable to execute required tool " + exc.getToolName();
+                  default -> "Service unavailable: please try again";
```

Comment:
Maybe here we should document the RuntimeException case. Saying that we recommend a catch all, for example.